core/modules/wcontrol/TransmitMgr.h

```diff
@@ -45,6 +45,9 @@ class QidMgr;
 class LockCount {
 public:
     LockCount() = default;
+    LockCount(LockCount const&) = delete;
+    LockCount& operator=(LockCount const&) = delete;
+    ~LockCount() = default;
 
     friend QidMgr;
```

Comment:
This header has 2 "friends". Too many "friends" is a sign of the "spaghetti" code since this is a way to bypass the `private` protection barrier to get uncontrolled access to the class's data members. Perhaps, this class (`LockCount`) could be moved into either the CC file or into the private section of its only user - the class `QidMgr`?. Then all members of `LockCount` could be made public?

Reply:
Nested classes is fine in this case. I'll make the change.